### PR TITLE
config.strictDepsByDefault: init option

### PR DESCRIPTION
**Pull request by Artturin**
so i won't have to change make-derivation.nix when i wan't to make this
true

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mindavi** [comment]
Somehow I really thought this already existed. I think this is useful to have to reduce the pain on the 'flag day' when we enable strictDeps by default (which would be really good to do).

Note that some hooks don't work with strictDeps, IIRC (needs reference to issue). So there may be some things to resolve.

**Comment by Artturin** [comment]
> Somehow I really thought this already existed.

that's what i think everytime i do a strictDeps by default run


> Note that some hooks don't work with strictDeps, IIRC (needs reference to issue). So there may be some things to resolve.

https://github.com/NixOS/nixpkgs/issues/56943

**Review by roberth — APPROVED** [review]
YES! Great 'first' step :100:

**Comment by Ericson2314** [comment]
BTW, mind adding the documentation for `contentAddressedByDefault` by default too?

File: pkgs/top-level/config.nix [hunk 1]
```diff
@@ -32,6 +32,10 @@ let
       feature = "run <literal>checkPhase</literal> by default";
     };
 
+    strictDepsByDefault = mkMassRebuild {
+      feature = "set <literal>strictDeps</literal> to true by default";
```
[the remark this answers is not in the document]
Comment (Artturin) [reply pkgs/top-level/config.nix:36]: these aren't rendered anywhere.

strictDeps isn't documented properly anywhere and it shouldn't be documented here but in the stdenv chapter in the manual

[the remark this answers is not in the document]
Comment (roberth) [reply pkgs/top-level/config.nix:36]: Ok. Two other, separate problems.

**Review by Ericson2314 — APPROVED** [review]
Looks good to me! Thanks for doing this!

**Pull request merged**